modules/chapter1/pages/section2.adoc

```diff
@@ -529,4 +544,155 @@ We're not going to go through the code and configuration for this, but here is t
+
+== Data Science Pipeline - Hands On Example
+
+We're now going to deploy and run an example DSP pipeline. The use case is offline scoring for _Fraud Detection_.
+
+=== Setup
+
+[%interactive]
+* [ ] Using the Minio UI, create a new S3 Bucket named _fraud_detection_bucket_.
+
+image::create-s3-fraud-bucket.png[]
+
+[%interactive]
+* [ ] Download the following artifacts:
+
+** [ ] xref:attachment$offline_scoring_kfp_pipeline.py[ONNX Model]
+** [ ] xref:attachment$live-data.csv.gz[Live Transaction Data]
+** [ ] xref:attachment$training-data.csv.gz[Model Training Data]
+
+[%interactive]
+* [ ] Uncompress the *_.gz_* files using the _gzip -d _ command. 
+* [ ] Upload the files to S3 _fraud_detection_bucket_ via the _Minio_ UI.
+
+====
+image::fraud-upload-files.png[]
+====
+====
+image::fraud-files-uploaded.png[]
+====
+
+[%interactive]
+* [ ] Create the *_aws-connection-fraud-detection_* _secret_ containing the _S3_ bucket name, endpoint and credentials.
+
+[source,bash]
+--
+oc create secret generic aws-connection-fraud-detection --from-literal=AWS_ACCESS_KEY_ID=minio --from-literal=AWS_DEFAULT_REGION=rhods-enablement-emea --from-literal=AWS_S3_BUCKET=fraud-detection-bucket --from-literal=AWS_S3_ENDPOINT=https://minio-api... --from-literal=AWS_SECRET_ACCESS_KEY=minio123
+--
+
+[%interactive]
+* [ ] Create two OpenShift Persistent Volumes with the following names:
+** *offline-scoring-model-volume*
+** *offline-scoring-data-volume*
+
+TIP: Example PVC definitions are here, download and customise as necessary this xref:attachment$offline-scoring-data-pvc.yaml[Data Volume definition] & xref:attachment$offline-scoring-model-pvc.yaml[Model Volume definition]
+
+
+=== Execute Pipeline
+
+Download and customise as necessary this xref:attachment$offline_scoring_kfp_pipeline.py[Example Volume definition]
+
+[%interactive]
+* [ ] Create a _Workbench_ with the name _fraud-onnx_
+
+image::fraud-onnx-workbench.png[]
+
+[%interactive]
+* [ ] Download the xref:attachment$offline_scoring_kfp_pipeline.py[Offline Scoring Pipeline] and load it into your _workbench_.
+* [ ] Click 'Restart Kernel and Run All Cells'.
+
+image::start-pipeline.png[]
+
+The Pipeline will be submitted to the server and a run _Triggered_.
+
+image::pipeline-run-triggered.png[]
+
+If the pipeline run was executed without issue you should see something similar to below:
+
+image::pipeline-finish.png[]
+
+=== Handling Pipeline Errors
+
+In the case that the pipeline failed the UI will be show something similar to the following:
+
+image::pipeline-failure.png[]
+
+Clicking on the *_failed_* node in the Graph and then clicking on the _Details_ tab will present the status of the execution step. 
+When the mouse is placed over the *_Status_* field the command to retrieve the detailed logs of the execution step is displayed.
+
+image::pipeline-error-details.png[]
+
+The step logs can then be viewed using the command provided.
```

Comment:
Optional comment could be to point the student to the Tekton pipeline run console to view the task logs.